crates/api/src/types.rs

```diff
@@ -392,37 +392,14 @@ impl MemoryType {
 /// imported from as well as the type of item that's being imported.
 #[derive(Debug, Clone, Hash, Eq, PartialEq)]
 pub struct ImportType {
-    module: String,
-    name: String,
-    ty: ExternType,
-}
-
-impl ImportType {
-    /// Creates a new import descriptor which comes from `module` and `name` and
-    /// is of type `ty`.
-    pub fn new(module: &str, name: &str, ty: ExternType) -> ImportType {
-        ImportType {
-            module: module.to_string(),
-            name: name.to_string(),
-            ty,
-        }
-    }
+    /// The module of the import.
+    pub module: String,
```

Comment:
I personally don't think we should have public-structs-with-public-fields in the API. They're quite inflexible in terms of future additions as well as changing the makeup of the internals. For example committing to two `String` allocations here I think may be something we want to tweak in the future. We could, for example, have something like `Arc<str>` which is cheaply cloneable and contains module/string, along with an index of where to split the string.

Reply:
I started cleaning this, and ended up finding it made sense to go ahead and eliminate the allocations. So we now have accessor functions, and they return `& 'module str`.